src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -244,6 +255,45 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
         });
     }
 
+    private async getControllerForInteractiveWindow(): Promise<string | undefined> {
+        // Fetch the active interpreter and use the matching controller
+        const api = await this.pythonApi.getApi();
+        const activeInterpreter = await api.getActiveInterpreter();
+        const interpreterControllers = this.notebookControllerManager.registeredNotebookControllers().filter(
+            (controller) =>
+                // We register each of our kernels as two controllers
+                // because controllers are currently per-viewtype. Find
+                // the one for the interactive viewtype for now
+                controller.controller.notebookType === InteractiveWindowView
+        );
+
+        // First try to match on the kernelSpec name. This is supposed to uniquely
+        // identify the active interpreter.
+        const kernelSpecName = getInterpreterKernelSpecName(activeInterpreter);
+        let preferredController = interpreterControllers.find((controller) => {
+            return (
+                controller.connection.kind === 'startUsingPythonInterpreter' &&
+                controller.connection.kernelSpec.name === kernelSpecName
```

Comment:
I don't think we're matching on this anymore. The name is essentially the 'path' to the kernelspec in the jupyter kernelspec folder. I mean this would match in the old code, but now this should always be python3 AFAIK.

The match you have after this seems like a better match. But it should include path and name, not just path.

Reply:
@rchiodo I pointed Joyce this direction here, not sure if I'm pointing things wrong. But here we are matching against an active interpreter. In the kernel finder code we getInterpreterKernelSpecName in the path to create kernelspecs to use for installed interpreters. So this generated name seemed like the safest match.

Reply:
Might be because the interpreter that I have is registered via the old style? Looking now.

Reply:
Yeah the old style used to generate a goofy hash for this name. We hopefully don't do this anymore.

Reply:
@rchiodo This is the kernel connection metadata for an new conda env that I just created:
![image](https://user-images.githubusercontent.com/812783/123686505-b8fd2a00-d804-11eb-8ab9-2490d79f6e81.png)
I think the hash is still there, but I'm looking at the implementation now, and I'm not sure if what I recommended is the right thing. The InterpreterKernelSpec name might just use the path for calculation of the name.